src/nvim/eval.c

```diff
@@ -7023,11 +7023,12 @@ static char *make_expanded_name(const char *in_start, char *expr_start, char *ex
 
   char *temp_result = eval_to_string(expr_start + 1, false);
   if (temp_result != NULL) {
-    retval = xmalloc(strlen(temp_result) + (size_t)(expr_start - in_start)
+    const size_t tr_len = strlen(temp_result);
+    retval = xmalloc(tr_len + (size_t)(expr_start - in_start)
                      + (size_t)(in_end - expr_end) + 1);
-    STRCPY(retval, in_start);
-    STRCAT(retval, temp_result);
-    STRCAT(retval, expr_end + 1);
+    char *p = STPCPY(retval, in_start);
+    p = STPCPY_LEN(p, temp_result, tr_len);
+    STRCPY(p, expr_end + 1);
```

Comment:
Why not use `xstrlcat`? That is what we currently recommend in clint.py: https://github.com/neovim/neovim/blob/2c9f22e7e4947e1865ab18c61fbc0199be8b323d/src/clint.py#L2489-L2492

btw should we add STRCAT to that lint rule?